libp2p-xtra/src/verify_peer_id.rs

```diff
@@ -0,0 +1,208 @@
+use crate::multiaddress_ext::MultiaddrExt;
+use futures::future::BoxFuture;
+use futures::stream::BoxStream;
+use futures::FutureExt;
+use futures::StreamExt;
+use futures::TryFutureExt;
+use futures::TryStreamExt;
+use libp2p_core::transport::ListenerEvent;
+use libp2p_core::transport::TransportError;
+use libp2p_core::Multiaddr;
+use libp2p_core::PeerId;
+use libp2p_core::Transport;
+use std::fmt;
+use std::fmt::Debug;
+
+/// A [`Transport`] wrapper that ensures we only connect to the expected peer.
+///
+/// Multi-addresses can contain a `/p2p` segment that specifies, which peer we want to connect to,
```

Comment:
🙃 Writing nit-pick. I'm pretty certain you should not separate the verb from its subject with a comma. Random source: https://www.grammarly.com/blog/comma-with-subjects-and-verbs/.
```suggestion
/// Multi-addresses can contain a `/p2p` segment that specifies which peer we want to connect to,
```

This pause can appear orally of course.